carp.deployment.core/src/commonMain/kotlin/dk/cachet/carp/deployment/domain/StudyDeployment.kt

```diff
@@ -34,6 +34,9 @@ class StudyDeployment( val protocolSnapshot: StudyProtocolSnapshot, val id: UUID
         data class DeviceUnregistered( val device: AnyDeviceDescriptor ) : Event()
         data class DeviceDeployed( val device: AnyMasterDeviceDescriptor ) : Event()
         data class DeploymentInvalidated( val device: AnyMasterDeviceDescriptor ) : Event()
+        // TODO: Immutable base class does not allow this to be defined as `object Stopped : Event()`.
+        //       It requires a data class, but that does not make sense since an object would still be immutable.
+        data class Stopped( val ignoreThis: Unit = Unit ) : Event()
```

Comment:
This is ugly and what I intended to possibly fix in this PR. But, then again, to consume this could you would just typecheck for `Stopped`. Removing the `ignoreThis` won't impact the consumer, so I guess I can easily fix this in the next release.

Reply:
Maybe `ignoreThis` should be private? Then the consumer cant access it.

Reply:
Remove the Immutable base class then. There's no point in having it here imho

Reply:
It is defined in the main DDD service interface. It makes sense to enforce immutable events in this pattern. I considered removing it, but then I have to remove it everywhere so I didn't. I'll look at it after release.

Reply:
But isn't the sealed class - data class pattern immutable in nature? I get why you use Immutable in the context of extendability and your serializer, but here it just seems...weird :)